services/catarse/lib/tasks/cron.rake

```diff
@@ -252,6 +256,10 @@ namespace :cron do
       CreateProjectFiscalToProjectSubAction.new(project_id: project.id, month: Time.zone.now.month, year: Time.zone.now.year).call
     end
 
+    Project.left_outer_joins(:project_fiscals).where(mode: %i[sub], state: %i[successful]).each do |project|
+      CreateProjectFiscalToProjectSubAction.new(project_id: project.id, month: Time.zone.now.month, year: Time.zone.now.year).call
+    end
+
     Project.joins(:project_fiscals).where(mode: %i[sub], state: %i[successful])
```

Comment:
**Problema/Dúvida**
Essa query aqui irá criar project fiscal duplicada para os mesmos projetos.
Nesse primeiro bloco, a consulta retornará todos os projetos com `mode == 'sub'` e com `state == 'successful`', pois esse left outer join não está sendo utilizado para filtrar nada, se removê-lo, o resultado da query será o mesmo e daí será criado um projeto fiscal para cada um dos projetos que atenda ao filtro.

Logo no bloco seguinte, a consulta tem um inner join com project fiscal e agora sim tem um filtro usando a tabela project_fiscals, mas nesse caso os projetos que essa consulta retornar, também estavam presentes na consulta do bloco anterior e vai criar de novo project fiscal pra esses projetos.

É isso mesmo ou tem alguma trava que impedirá a criação de project fiscal duplicado?

Reply:
Oi @stephannv, erro meu. Ficou falando um "where", nesse commit. A ideia é usar assim:
`Project.left_outer_joins(:project_fiscals).where('project_fiscals.id IS NULL')` capturando todos os projetos que não tenham project_fiscals associados. 
Testei no console e estava funcionando corretamente. Subi um novo commit.